File: ros2_ws/src/my_py_pkg/my_py_pkg/number_publisher.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64
from my_custom_interfaces.msg import HardwareStatus

#Creating new class that inherits from the rclpy Node class
class NumberPublisherNode(Node):
    def __init__(self):
        super().__init__('number_publisher')
        self.number_ = 2
        self.number_publisher_ = self.create_publisher(Int64, 'number', 10)
        self.hardware_status_publisher_ = self.create_publisher(HardwareStatus, 'hardware_status', 10)
        self.number_timer_ = self.create_timer(1.0, self.publish_number)
        self.get_logger().info('Number Publisher Node has been started.')

    def publish_number(self):
        msg = Int64()
        msg.data = self.number_
        self.number_publisher_.publish(msg)
    
    def publish_hardware_status(self):
        msg = HardwareStatus()
        msg.version = 1
        msg.temperature = 36.5
        msg.are_motors_ok = True
        msg.debug_message = "All systems are functional."
        self.hardware_status_publisher_.publish(msg)

def main(args=None):
    #Initialize communications
    rclpy.init(args=args)
    #Create object from NumberPublisherNode class
    node = NumberPublisherNode()
    #make node spin, this will block execution, the program and node stays alive, all registered callbacks will be processed
    rclpy.spin(node)
    #after node is killed, shutdown communication
    rclpy.shutdown()
    
if __name__ == '__main__':
    main()

File: ros2_ws/src/my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"
#include "my_custom_interfaces/msg/hardware_status.hpp"

class NumberPublisherNode : public rclcpp::Node
{
public:
    NumberPublisherNode() : Node("number_publisher")
    {
        number_publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        hardware_status_publisher_ = this->create_publisher<my_custom_interfaces::msg::HardwareStatus>("hardware_status", 10);
        number_timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&NumberPublisherNode::publishNumber, this));
        hardware_status_timer_ = this->create_wall_timer(std::chrono::seconds(2), std::bind(&NumberPublisherNode::publishHardwareStatus, this));
        RCLCPP_INFO(this->get_logger(), "Number publisher has been started.");
    }
    void publishNumber()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = number_;
        number_publisher_->publish(msg);
    }
    void publishHardwareStatus()
    {
        auto msg = my_custom_interfaces::msg::HardwareStatus();
        msg.version = 1;
        msg.temperature = 69.69;
        msg.are_motors_ok = true;
        msg.debug_message = "All systems are functional.";
        hardware_status_publisher_->publish(msg);
    }
private:
    int number_ = 69;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr number_publisher_;
    rclcpp::Publisher<my_custom_interfaces::msg::HardwareStatus>::SharedPtr hardware_status_publisher_;
    rclcpp::TimerBase::SharedPtr number_timer_;
    rclcpp::TimerBase::SharedPtr hardware_status_timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}